rclpy/rclpy/qos.py

```diff
@@ -12,8 +12,114 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from enum import IntEnum
+import rclpy
+
 
 class QoSProfile:
-    pass
+    """Class defining Quality of Service policies."""
+
+    __slots__ = [
+        '_history',
+        '_depth',
+        '_reliability',
+        '_durability',
+    ]
+
+    def __init__(self, **kwargs):
+        assert all(['_' + key in self.__slots__ for key in kwargs.keys()]), \
+            "Invalid arguments passed to constructor: %r" % kwargs.keys()
+        self.history = kwargs.get(
+            'history',
+            QoSHistoryPolicy.RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT)
+        self.depth = kwargs.get('depth', int())
+        self.reliability = kwargs.get(
+            'reliability',
+            QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_SYSTEM_DEFAULT)
+        self.durability = kwargs.get(
+            'durability',
+            QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT)
+
+    @property
+    def history(self):
+        """Get QoSHistoryPolicy 'history'."""
+        return self._history
+
+    @history.setter
+    def history(self, value):
+        assert isinstance(value, QoSHistoryPolicy)
+        self._history = value
+
+    @property
+    def reliability(self):
+        """Get QoSReliabilityPolicy 'reliability'."""
+        return self._reliability
+
+    @reliability.setter
+    def reliability(self, value):
+        assert isinstance(value, QoSReliabilityPolicy)
+        self._reliability = value
+
+    @property
+    def durability(self):
+        """Get QoSDurabilityPolicy 'durability'."""
+        return self._durability
+
+    @durability.setter
+    def durability(self, value):
+        assert isinstance(value, QoSDurabilityPolicy)
+        self._durability = value
+
+    @property
+    def depth(self):
+        """Get int 'depth'."""
+        return self._depth
+
+    @depth.setter
+    def depth(self, value):
+        assert isinstance(value, int)
+        self._depth = value
+
+    def get_QoS_Profile(self):
```

Comment:
fixed function name in https://github.com/ros2/rclpy/commit/93503d0be408db4d4906023bd0f062c5ce05a45a